### PR TITLE
feat(scripts/develop.sh): add --debug flag to develop.sh

### PR DESCRIPTION
**Pull request by johnstcn**
Adds a `--debug` flag to `scripts/develop.sh` that will start coder under `dlv debug` instead.
You can then use e.g. the following launch snippet to connect dlv:
```
    {
      "name": "Delve Remote",
      "type": "go",
      "request": "attach",
      "mode": "remote",
      "port": 12345,
    }
```

You can also run invididual CLI commands under dlv e.g.

```
DEBUG_DELVE=1 scripts/coder-dev.sh list
```

### CODE REVIEW

File: scripts/coder-dev.sh [hunk 1]
```diff
@@ -55,4 +56,11 @@ coder)
...
 esac
 
-exec "${CODER_DEV_BIN}" --global-config "${CODER_DEV_DIR}" "$@"
+runcmd="${CODER_DEV_BIN}"
+if [[ "${debug}" == 1 ]]; then
+	set -x
+	runcmd="dlv debug --headless --continue --listen 127.0.0.1:12345 --accept-multiclient ./cmd/coder --"
+fi
+
+# shellcheck disable=SC2086 # This is probably one of the few times you actually want this.
```
Comment (mafredri) [inline scripts/coder-dev.sh:65]: Feel free to change or not change, just pointing out the option:

Alternatively, we could make `runcmd` and array (`runcmd=(dlv debug ...)`) and use `"${runcmd[@]}"` (quotes are OK here due to @, it essentially produces `"$1" "$2"` vs `$1 $2`).

File: scripts/coder-dev.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ source "${SCRIPT_DIR}/lib.sh"
 
 GOOS="$(go env GOOS)"
 GOARCH="$(go env GOARCH)"
+debug="${debug:-0}"
```
Comment (mafredri) [inline scripts/coder-dev.sh:13]: Wondering if this name is too common for a shell var that changes behavior? Maybe `DEBUG_DELVE` or something similar like that?

**Review by mafredri — APPROVED** [review]

File: scripts/coder-dev.sh (hunk 1 above)
Comment (johnstcn) [reply scripts/coder-dev.sh]: Added!

File: scripts/coder-dev.sh (hunk 2 above)
Comment (johnstcn) [reply scripts/coder-dev.sh]: Done!

**Review by dannykopping — APPROVED** [review]
Thanks a tonne!

**Pull request merged**